Dockerfile

```diff
@@ -1,22 +1,41 @@
 # Base will install runtime dependencies and configure generics
-FROM node:12.16
+FROM node:12-slim as base
+LABEL maintainer="Marko Kajzer <markokajzer91@gmail.com>, Nico Stapelbroek <discord-soundbot@nstapelbroek.com>"
+
+RUN apt-get update -q && apt-get install -qy wget
+RUN wget -qO /tini https://github.com/krallin/tini/releases/download/v0.18.0/tini-$(dpkg --print-architecture) && \
+    chmod +x /tini && mkdir /app && chown -R node:node /app
 
-LABEL maintainer="Marko Kajzer <markokajzer91@gmail.com>"
 WORKDIR /app
 
-# Copy some files
-COPY package.json yarn.lock ./
-COPY . /app
 
-# Install ffmpeg and other deps
-RUN apt-get update
-RUN apt-get install --yes build-essential git ffmpeg make python > /dev/null
-RUN yarn install --silent
+# Builder will install system dependencies
+FROM base as builder
```

Comment:
Do we need two stages for building? Looks _to me_ that this could be merged with `build`.

Reply:
I wanted to create an image stage that you could potentially publish and reuse as a build environment.
Some CI's like [Gitlab](https://docs.gitlab.com/ee/ci/docker/using_docker_images.html) and [Circle CI](https://circleci.com/docs/2.0/custom-images/) allow you to bring your own builder image to speed up CI. 

This also creates a clear separation between what you as a developer should have installed to develop vs what commands to run for building your project. I admit it's a bit of a personal nitpick, so If you want to I can combine them 😄

Reply:
Interesting! Thanks for your insight, let's keep it as is :)